amgut/lib/data_access/redcap.py

```diff
@@ -0,0 +1,115 @@
+from urllib import urlencode
+from json import loads
+
+from tornado.web import HTTPError
+from tornado.httpclient import HTTPClient
+
+from amgut.lib.config_manager import AMGUT_CONFIG
+from amgut.lib.data_access.sql_connection import TRN
+
+
+def _make_request(data):
+    body = urlencode(data)
+    client = HTTPClient()
+    response = client.fetch(AMGUT_CONFIG.redcap_url, method='POST',
+                            headers=None, body=body)
+    # Response not always JSON, so can't always use loads
+    if '{"error"' in response.body:
+        raise HTTPError(400, loads(response.body)['error'])
+    return response.body
+
+
+def get_survey_url(record, instrument='ag_human_en_us'):
+    """Genereates a new survey link for a given record
+
+    Parameters
+    ----------
+    record : int
+        Redcap record (consent) to attach this survey to
+    instrument : str, optional
+        The instrument (survey) to get the URL for. Default ag_human_en_us
+
+    Returns
+    -------
+    str
+        URL for the survey
+    """
+    with TRN:
```

Comment:
Thanks!

On Mon, Apr 25, 2016 at 12:18 PM, Joshua Shorenstein <
notifications@github.com> wrote:

> In amgut/lib/data_access/redcap.py
> <https://github.com/biocore/american-gut-web/pull/579#discussion_r60971545>
> :
>
> > +def get_survey_url(record, instrument='ag_human_en_us'):
> > +    """Genereates a new survey link for a given record
> > +
> > +    Parameters
> > +    ----------
> > +    record : int
> > +        Redcap record (consent) to attach this survey to
> > +    instrument : str, optional
> > +        The instrument (survey) to get the URL for. Default ag_human_en_us
> > +
> > +    Returns
> > +    -------
> > +    str
> > +        URL for the survey
> > +    """
> > +    with TRN:
>
> I can do an explanation at the top of how redcap breaks down their IDs and
> other pertinent information. I'll add that now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/american-gut-web/pull/579/files/6afc11020fe2322bdd9b15fd83fd0a431feb0f43#r60971545>
>